mmv1/products/appengine/DomainMapping.yaml

```diff
@@ -78,7 +76,7 @@ properties:
     name: 'sslSettings'
     description: |
       SSL configuration for this domain. If unconfigured, this domain will not serve with SSL.
-    diff_suppress_func: 'sslSettingsDiffSuppress'
+    default_from_api: true
```

Comment:
OK- the issue we run into is that we get two different responses depending on the API version.

```
 "sslSettings": {
  "sslManagementType": "AUTOMATIC"
 }
```

is returned as the following in GA

```
  "sslSettings": {
    "sslManagementType": "AUTOMATIC",
    "pendingManagedCertificateId": "34199778"
  },
```

And the following in Beta

```
  "sslSettings": {
   "pendingManagedCertificateId": "34199778"
  },
```

I _suspect_ that `MANUAL` will still be correctly returned, but my test project was having issues so I wasn't able to verify. We may need to amend things if it doesn't get returned as well- the only test that covers it is the second stage of the _update test failing on the first stage so far.

---

To resolve this we'll want to introduce a decoder (https://googlecloudplatform.github.io/magic-modules/develop/custom-code/#decoder), similar to something like https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/templates/terraform/decoders/disk.erb. In decoders we have access to `res`, a `map[string]interface{}` representing the JSON response to Get.

In there you'll want to check if sslSettings is set. If it is, open its contents (Just like the `original` fields in the Disk sample I linked above). If sslManagementType is not set and pendingManagedCertificateId is, add the field sslManagementType with the value "AUTOMATIC". Then, just return the modified object.

Effectively, we're just rewriting the 2nd response into the first response in that code, which is what we already expect to handle.

Reply:
I believe it's done ^^

Reply:
Looks like it to me- running tests.